## 1. Introducción a la estadística descriptiva

La estadística descriptiva es una parte crucial de cualquier análisis de datos. Nos ayuda a entender la distribución de los datos, identificar patrones y tendencias, y resumir los datos de una manera que sea fácil de entender. Los dos conceptos clave que discutiremos en este notebook son el promedio y la mediana.

## 2. Promedio y su interpretación

El promedio, también conocido como la media, es una medida estadística que se utiliza para entender el centro de una distribución de datos. Se calcula sumando todos los números en el conjunto de datos y luego dividiendo por la cantidad de números en el conjunto de datos.

Aunque el promedio es una herramienta útil para resumir un conjunto de datos, puede ser engañoso si el conjunto de datos contiene valores extremos. Esto se debe a que el promedio es sensible a los valores extremos, lo que significa que puede ser fuertemente influenciado por los valores que son significativamente más altos o más bajos que el resto.

## 3. Mediana y su interpretación

La mediana es otra medida estadística que se utiliza para entender el centro de una distribución de datos. A diferencia del promedio, la mediana es el valor medio en un conjunto de datos cuando los números están ordenados en orden ascendente o descendente.

La mediana es una medida resistente, lo que significa que no es sensible a los valores extremos. Por lo tanto, en un conjunto de datos con valores extremos, la mediana puede ser una mejor representación del centro de la distribución de datos.

## 4. Comparación entre promedio y mediana

Tanto el promedio como la mediana son medidas de tendencia central que se utilizan para entender el centro de una distribución de datos. Sin embargo, tienen diferentes propiedades:

- **Promedio:** Es útil cuando los datos están uniformemente distribuidos y no hay valores extremos. Sin embargo, puede ser engañoso si hay valores extremos en los datos.
- **Mediana:** Es útil cuando los datos tienen valores extremos que podrían sesgar el promedio. La mediana es resistente a los valores extremos y puede proporcionar una mejor representación del centro de la distribución de datos en tales casos.

En la práctica, es útil calcular tanto el promedio como la mediana y compararlos para entender la distribución de los datos.

## 5. Ejemplo: Efecto de Bill Gates en un bar

Para ilustrar cómo el promedio puede ser engañoso y cómo la mediana puede ser una mejor representación en ciertos casos, consideremos el siguiente ejemplo:

Imagina que Bill Gates, uno de los hombres más ricos del mundo, entra en un bar. El ingreso promedio de las personas en el bar aumentaría drásticamente. Sin embargo, esto no significa que todos en el bar sean repentinamente más ricos. Este es un ejemplo de cómo el promedio puede ser engañoso.

Por otro lado, si calculamos la mediana de los ingresos, veremos que no se ve afectada por la entrada de Bill Gates. La mediana seguiría siendo la misma, proporcionando una mejor representación de los ingresos de la mayoría de las personas en el bar.

A continuación, vamos a simular este escenario con código Python.

In [ ]:
# Import necessary libraries
import numpy as np

# Let's assume there are 10 people in the bar and each of them earns $100
incomes = np.repeat(100, 10)

# Calculate the average income
average_income = np.mean(incomes)
print(f'Average income before Bill Gates enters the bar: {average_income}')

# Calculate the median income
median_income = np.median(incomes)
print(f'Median income before Bill Gates enters the bar: {median_income}')

# Now, let's assume Bill Gates enters the bar. His net worth is around $100 billion
incomes = np.append(incomes, 100_000_000_000)

# Calculate the new average income
average_income = np.mean(incomes)
print(f'Average income after Bill Gates enters the bar: {average_income}')

# Calculate the new median income
median_income = np.median(incomes)
print(f'Median income after Bill Gates enters the bar: {median_income}')

Average income before Bill Gates enters the bar: 100.0
Median income before Bill Gates enters the bar: 100.0
Average income after Bill Gates enters the bar: 9090909181.818182
Median income after Bill Gates enters the bar: 100.0


Como podemos ver en los resultados, el ingreso promedio aumentó drásticamente de $100 a $9,090,909,181.82 cuando Bill Gates entró en el bar. Sin embargo, la mediana del ingreso se mantuvo en $100, lo que indica que la mayoría de las personas en el bar todavía ganan $100. Esto ilustra cómo el promedio puede ser engañoso en presencia de valores extremos y cómo la mediana puede ser una mejor representación en tales casos.

## 6. Conclusión

En este notebook, hemos discutido los conceptos de promedio y mediana en la estadística descriptiva. Hemos visto cómo el promedio puede ser engañoso en presencia de valores extremos y cómo la mediana puede ser una mejor representación en tales casos. También hemos ilustrado estos conceptos con el ejemplo de Bill Gates entrando en un bar.

Es importante recordar que tanto el promedio como la mediana son herramientas útiles en estadística descriptiva y cada una tiene sus propios usos y limitaciones. La elección de usar el promedio o la mediana depende de la distribución de los datos y del contexto del análisis.